src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExpressionBinder.cs

```diff
@@ -1160,17 +1155,15 @@ private Expr UnwrapExpression(Expr pExpression)
 
         private static ErrorCode GetStandardLvalueError(CheckLvalueKind kind)
         {
+            Debug.Assert(Enum.IsDefined(typeof(CheckLvalueKind), kind));
             switch (kind)
             {
-                default:
-                    VSFAIL("bad kind");
-                    return ErrorCode.ERR_AssgLvalueExpected;
-                case CheckLvalueKind.Assignment:
-                    return ErrorCode.ERR_AssgLvalueExpected;
                 case CheckLvalueKind.OutParameter:
                     return ErrorCode.ERR_RefLvalueExpected;
                 case CheckLvalueKind.Increment:
                     return ErrorCode.ERR_IncrementLvalueExpected;
+                default:
+                    return ErrorCode.ERR_AssgLvalueExpected;
```

Comment:
I mean, the one you added is not equivalent. Oh, I see @stephentoub gave basically the same feedback.